src/providers/ad/ad_common.c

```diff
@@ -107,26 +114,149 @@ ad_create_default_sdap_options(TALLOC_CTX *mem_ctx)
     return NULL;
 }
 
+static errno_t
+ad_create_sdap_options(TALLOC_CTX *mem_ctx,
+                       struct confdb_ctx *cdb,
+                       const char *conf_path,
+                       struct sdap_options **_id_opts)
+{
+    struct sdap_options *id_opts;
+    errno_t ret = EOK;
+
+    if (cdb == NULL || conf_path == NULL) {
```

Comment:
Can this ever happen? The way I read the code, we always pass the confdb pointer from the be_ctx structure where I don't think it can be NULL and the conf_path is also either returned from be_ctx or from create_subdom_conf_path whose result is checked.